packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -122,11 +123,13 @@ export class SelectedNetworkController extends BaseController<
    * @param options.messenger - The restricted controller messenger for the EncryptionPublicKey controller.
    * @param options.state - The controllers initial state.
    * @param options.getUseRequestQueue - feature flag for perDappNetwork & request queueing features
+   * @param options.domainProxyMap - an object that implements the Map interface for Domain to NetworkProxy
```

Comment:
Nit: We should provide more guidance here on what our expectations are about this parameter. We don't need to talk about it implementing the Map interface either, the type does that for us.

Maybe something like

```suggestion
   * @param options.domainProxyMap - A map for storing domain-specific proxies that are held in memory only during use.
```